test/request.go

```diff
@@ -163,11 +176,19 @@ func WaitForEndpointStateWithTimeout(
 		return nil, err
 	}
 
+	var tOpts []spoof.TransportOption
 	for _, opt := range opts {
-		opt(req)
+		rOpt, ok := opt.(RequestOption)
```

Comment:
``
if r, ok := opt.(RO); ok {
  r(req)
} else if r, ok := opt.(s.TO); ok {
  to = append(to, r)
}
```
It's one or the other, so if else or continue.